### PR TITLE
[kube-prometheus-stack] Update dependencies / prometheus-operator 0.47.1

### PR DESCRIPTION
**Pull request by mrueg**
#### What this PR does / why we need it:
* Bump to kube-state-metrics v2 which deprecated some metrics and changed
CLI arguments which might cause breakage. See: https://kubernetes.io/blog/2021/04/13/kube-state-metrics-v-2-0/

* Update Prometheus-Operator to 0.47.1

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by amuraru** [comment]
the upgrade to ksmv2 also breaks the canned grafana dashboards. we should also look into those and ensure those get an update as well

**Comment by mrueg** [comment]
> the upgrade to ksmv2 also breaks the canned grafana dashboards. we should also look into those and ensure those get an update as well

good catch. let me take a look and see what needs to be changed there.

**Comment by mrueg** [comment]
it looks like https://github.com/prometheus-community/helm-charts/pull/966 updates the dashboards already.

**Comment by amuraru** [comment]
for clarity i would merge #966 into this PR

**Comment by invidian** [comment]
>for clarity i would merge #966 into this PR

Please go ahead at your convenience :)

File: charts/kube-prometheus-stack/README.md [hunk 1]
```diff
@@ -83,6 +83,9 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 
 A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an incompatible breaking change needing manual actions.
 
+### From 15.x to 16.x
+Version 16 upgrades kube-state-metrics to v2.0.0. This includes changed command-line arguments and removed metrics, see this [blog post](https://kubernetes.io/blog/2021/04/13/kube-state-metrics-v-2-0/).
+
```
Comment (invidian) [inline charts/kube-prometheus-stack/README.md:88]: BTW, list of CRDs to update below is missing link to PrometheusRules CRD.

Comment (mrueg) [reply charts/kube-prometheus-stack/README.md:88]: This should be okay, it hasn't changed in 6 months. In general I would assume we only support jumping one major release a time.

**Comment by mrueg** [comment]
> > for clarity i would merge #966 into this PR
> 
> Please go ahead at your convenience :)

Thanks, I did and removed the outdated grafana dashboards as well. They were only around to support k8s 1.14 and earlier which should be considered ancient as of now.

File: charts/kube-prometheus-stack/templates/grafana/configmap-dashboards.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {{- if and .Values.grafana.enabled .Values.grafana.defaultDashboardsEnabled }}
-{{- $files := .Files.Glob "dashboards/*.json" }}
+{{- $files := .Files.Glob "dashboards-1.14/*.json" }}
```
Comment (mrueg) [inline charts/kube-prometheus-stack/templates/grafana/configmap-dashboards.yaml:2]: reviewers, PTAL if this is correct.

File: charts/kube-prometheus-stack/README.md (hunk 1 above)
Comment (invidian) [reply charts/kube-prometheus-stack/README.md:88]: Make sense, thanks for explanation! I found it a bit confusing.

File: charts/kube-prometheus-stack/README.md [hunk 3]
```diff
@@ -83,6 +83,9 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 
 A major chart version change (like v1.2.3 -> v2.0.0) indicates that there is an incompatible breaking change needing manual actions.
 
+### From 15.x to 16.x
+Version 16 upgrades kube-state-metrics to v2.0.0. This includes changed command-line arguments and removed metrics, see this [blog post](https://kubernetes.io/blog/2021/04/13/kube-state-metrics-v-2-0/). This version also removes grafana dashboards that supported kubernetes 1.14 or earlier.
```
Comment (invidian) [inline charts/kube-prometheus-stack/README.md:87]:
```suggestion
Version 16 upgrades kube-state-metrics to v2.0.0. This includes changed command-line arguments and removed metrics, see this [blog post](https://kubernetes.io/blog/2021/04/13/kube-state-metrics-v-2-0/). This version also removes Grafana dashboards that supported Kubernetes 1.14 or earlier.
```

File: charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py [hunk 4]
```diff
@@ -32,25 +32,11 @@ def new_representer(dumper, data):
         'min_kubernetes': '1.14.0-0'
     },
     {
-        'source': 'https://raw.githubusercontent.com/etcd-io/website/master/content/docs/current/op-guide/grafana.json',
+        'source': 'https://raw.githubusercontent.com/etcd-io/website/master/content/en/docs/v3.4/op-guide/grafana.json',
         'destination': '../templates/grafana/dashboards-1.14',
         'type': 'json',
         'min_kubernetes': '1.14.0-0'
```
Comment (invidian) [inline charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:38]: I think we could remove the whole `min_kubernetes` and `max_kubernetes` logic all together then.

File: charts/kube-prometheus-stack/templates/grafana/configmap-dashboards.yaml (hunk 2 above)
Comment (invidian) [reply charts/kube-prometheus-stack/templates/grafana/configmap-dashboards.yaml:2]: I don't see any previous reference to `dashboards-1.14`. Was it never used then? :thinking:

File: charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py (hunk 4 above)
Comment (mrueg) [reply charts/kube-prometheus-stack/hack/sync_grafana_dashboards.py:38]: I would keep it, we might need it at some other point if things become incompatible again.

**Comment by ArgTang** [comment]
another state-metrics pull req #968 should decide on which to use.

**Comment by mrueg** [comment]
> another state-metrics pull req #968 should decide on which to work on, to not duplicate effort

this one here is about kube-prometheus-stack and not regular prometheus.

**Review by bismarck — APPROVED** [review]
lgtm

**Pull request merged**